### DISCUSSION TITLE
PushDelgate.OnTokenRefreshed not being ALWAYS called with 3.2.4

### DISCUSSION THREAD

**Discussion by munkii** [shinyorg/shiny#1407] [Q&A] [answered]
This is probably an non issue as Push Notifications seem to be working fine, but but previously  OnTokenRefreshed was called every time on Startup or at least that is what I observed when running a debug version of either Android or iOS.

This used to surprise me as it looks like it should only be called when the Token is refreshed. And the Shiny library says "This is called ONLY when the token changes, not during RequestAccess"

Was the previous calling of OnTokenRefreshed a bug that has been fixed in 3.x? Push Notifications certainly seem to be working fine.

**Comment by aritchie** [comment, marked as answer]
It is called whenever the token is changed (including RequestAccess if there is a change).  The docs need an update

**Reply by munkii** [reply]
I wasn't imagining it was I. In previous versions it got called a lot more often even when the token had not changed.
